sport-moment-nft/contracts/EPL.cdc

```diff
@@ -0,0 +1,1124 @@
+            // Future sets may include serial numbers, so leaving data structure in-tact
+            let momentNFT <- create NFT(
+                editionID: self.id,
+                serialNumber: 0
+            )
+            EPL.totalSupply = EPL.totalSupply + 1
+            // Keep a running total (you'll notice we used this as the serial number)
+            self.numMinted = self.numMinted + 1 as UInt64
+
+            return <- momentNFT
+        }
+
+        /// initializer
+        ///
+        init (
+            seriesID: UInt64,
+            setID: UInt64,
+            playID: UInt64,
+            maxMintSize: UInt64?,
+            tier: String,
+        ) {
+            pre {
+                maxMintSize != 0: "max mint size is zero, must either be null or greater than 0"
+                EPL.seriesByID.containsKey(seriesID): "seriesID does not exist"
+                EPL.setByID.containsKey(setID): "setID does not exist"
+                EPL.playByID.containsKey(playID): "playID does not exist"
+                EPL.getSeriesData(id: seriesID)!.active == true: "cannot create an Edition with a closed Series"
+                EPL.getSetData(id: setID)!.locked == false: "cannot create an Edition with a locked Set"
+            }
+
+            self.id = EPL.nextEditionID
+            self.seriesID = seriesID
+            self.setID = setID
+            self.playID = playID
+
+            // If an edition size is not set, it has unlimited minting potential
+            if maxMintSize == 0 {
+                self.maxMintSize = nil
+            } else {
+                self.maxMintSize = maxMintSize
+            }
+
+            self.tier = tier
+            self.numMinted = 0 as UInt64
+
+            EPL.nextEditionID = EPL.nextEditionID + 1 as UInt64
+
+            emit EditionCreated(
+                id: self.id,
+                seriesID: self.seriesID,
+                setID: self.setID,
+                playID: self.playID,
+                maxMintSize: self.maxMintSize,
+                tier: self.tier,
+            )
+        }
+    }
+
+    /// Get the publicly available data for an Edition
+    ///
+    pub fun getEditionData(id: UInt64): EditionData? {
+        if EPL.editionByID[id] == nil{
+            return nil
+        }
+
+        return EPL.EditionData(id: id)
+    }
+
+    // Validate tags exist
+    //
+    pub fun validateTags(tagIds: [UInt64]): Bool {
+        for tagId in tagIds {
+            if EPL.tagByID[tagId] == nil {
+                return false
+            }
+        }
+        return true
+    }
+
+    //------------------------------------------------------------
+    // NFT
+    //------------------------------------------------------------
+
+    /// A Moment NFT
+    ///
+    pub resource NFT: NonFungibleToken.INFT, MetadataViews.Resolver {
+        pub let id: UInt64
+        pub let editionID: UInt64
+        pub let serialNumber: UInt64
+        pub let mintingDate: UFix64
+
```

Comment:
thoughts on an extension field?
`pub let ext: {String: AnyStruct}`

Reply:
for when we want to add additional properties to the NFT without introducing a breaking change

Reply:
Yh, that's what attachments in cadence are for

Reply:
I don't agree. If an NFT project wants to directly add a new field to an NFT, it's much easier to use something already accessible than an attachment which design is not finalized yet. 
imagine if you want to implement locking, using this ext you can easily do that by adding a lockedField. what are the steps to do it with attachment

Reply:
Done. Good suggestion, Eric.